compositions/upbound-aws-provider/apigw/rest.yaml

```diff
@@ -0,0 +1,282 @@
+        - type: FromCompositeFieldPath
+          fromFieldPath: metadata.name
+          toFieldPath: metadata.name
+          transforms:
+            - type: string
+              string:
+                fmt: "%s-restapi"
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.apiDescription
+          toFieldPath: spec.forProvider.description
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.apiName
+          toFieldPath: spec.forProvider.name
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.openApiSpecification
+          toFieldPath: spec.forProvider.body
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.endpointType
+          toFieldPath: spec.forProvider.endpointConfiguration[0].types[0]
+        - type: ToCompositeFieldPath
+          fromFieldPath: status.atProvider.arn
+          toFieldPath: status.apigwArn
+        - type: ToCompositeFieldPath
+          fromFieldPath: status.atProvider.executionArn
+          toFieldPath: status.apiExecutionArn
+        - type: ToCompositeFieldPath
+          fromFieldPath: status.atProvider.id
+          toFieldPath: status.apiId
+        - type: ToCompositeFieldPath
+          fromFieldPath: status.atProvider.name
+          toFieldPath: status.apiName
+        - type: ToCompositeFieldPath
+          fromFieldPath: status.atProvider.executionArn
+          toFieldPath: status.accountId
+          transforms:
+            - type: string
+              string:
+                type: Regexp
+                regexp:
+                  match: \d{12,}          
+
+    - name: apigw-role
+      base:
+        apiVersion: iam.aws.upbound.io/v1beta1
+        kind: Role
+        spec:
+          deletionPolicy: Delete
+          forProvider:
+            assumeRolePolicy: |
+              {
+                "Version": "2012-10-17",
+                "Statement": [
+                  {
+                    "Effect": "Allow",
+                    "Principal": {
+                      "Service": "apigateway.amazonaws.com"
+                    },
+                    "Action": "sts:AssumeRole"
+                  }
+                ]
+              }
+            path: "/"
+            managedPolicyArns:
+              - >-
+                arn:aws:iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs
+      patches:
+        - type: PatchSet
+          patchSetName: common-fields
+        - type: FromCompositeFieldPath
+          fromFieldPath: metadata.name
+          toFieldPath: metadata.name
+          transforms:
+            - type: string
+              string:
+                fmt: "%s-apigw-role"
+        - type: ToCompositeFieldPath
+          fromFieldPath: status.atProvider.arn
+          toFieldPath: status.apigwRoleArn
+        - type: ToCompositeFieldPath
+          fromFieldPath: status.atProvider.id
+          toFieldPath: status.apigwRoleName
+
+    - name: api-deployment
```

Comment:
Does this actually re-deploy on code changes? Last time I played with API GW on Crossplane, it essentially ran into the same issue with CF where deployments are not updated with artifact changes.

Reply:
Terraform takes care of the deployments when API definition changes, similarly to SAM deploy command. I've tested by making changes in the composition (OpenAPI definition) - they get deployed and reflect in AWS Management Console in under minute or so.